src/main/kotlin/net/hatomg/Main.kt

```diff
@@ -0,0 +1,40 @@
+package net.hatomg
+
+import java.io.File
+import java.nio.file.Paths
+import kotlin.system.exitProcess
+
+fun main(args: Array<String>) {
+    if (args.isEmpty()) {
+        println("使用方法: backup_management-all.jar <ディレクトリ名>")
+        exitProcess(1)
+    }
+
+    val dirPath = args[0]
+    val dir = File(dirPath)
+
+    if (!dir.isDirectory) {
+        println("指定されたパスはディレクトリではありません: $dirPath")
+        exitProcess(1)
+    }
+
+    val currentExe = Paths.get(".").toAbsolutePath().normalize().toFile()
```

Comment:
```suggestion
    val currentExe = Paths.get(".").toAbsolutePath().normalize().toFile()
```
このコードは不要だと思うのですがどうなのでしょうか？

Reply:
そのコードを消すと jarファイル自身が削除の対象になる可能性があると思います。
対象の拡張子を指定しておくのが簡単かも？
あとは、誤爆対策として 実行者に削除するファイル名を表示して Y/N で一回確認取るコード追加してもいいかもです。

Reply:
Paths.get() で相対パスを指定すると良くない(別ディレクトリからjarが実行されると結果が変わる)ので、こうすれば実行場所に関わらず動くはずです。
軽く調べたものなので、デバッグお願いします!!
```suggestion
    val currentExe = File(Main::class.java.protectionDomain.codeSource.location.toURI()).toPath().toAbsolutePath().normalize().toFile()
```

Reply:
そのコードだとビルドは通りません。
```
Unresolved reference 'Main'.
Cannot infer type for this parameter. Please specify it explicitly.
```

Reply:
ここに問題はありませんでした。
```kotlin
File(Main::class.java.protectionDomain.codeSource.location.toURI()).toPath().toAbsolutePath().normalize().toFile()
```
に変更したとしても問題は解決しません。